include/slb/type_traits.hpp

```diff
@@ -905,25 +906,70 @@ struct is_convertible
     : slb::bool_constant<std::is_convertible<From, To>::value> {};
 #endif
 
-#if SLB_INTEGRAL_CONSTANT == 2 && __cpp_lib_is_invocable // C++17
+// We only enable the C++17 implementation under C++2a here to account for
+// P0704: "Fixing const-qualified pointers to members".
+
+#if SLB_INTEGRAL_CONSTANT == 2 && __cpp_lib_is_invocable /* C++17 */ &&        \
+    __cplusplus > 201703L /* C++2a */
 using std::is_invocable;
 using std::is_invocable_r;
 using std::is_nothrow_invocable;
 using std::is_nothrow_invocable_r;
 #else
-// TODO
+namespace detail {
+
+template <typename F, typename Enable = void>
+struct is_callable {
```

Comment:
Perhaps we can call these `invocable_traits`?

Reply:
Why? These are not the LFTS `callable_traits`.

Reply:
I'm mainly thinking about the fact that these are `is_invocable` implementation details,
rather than a utility serving a different concept of "callable". The fact that it provides
`value` and `nothrow` seemed to suggest that it's a trait utility where it takes a type
and provides answers to a few traits, rather than simply answering a boolean question.

Anyway, I think maybe rather than introducing "callable", we can just call these
`is_invocable_impl`, or something along those lines? Unless you feel that "callable"
is the correct term here?

Reply:
The names `is_callable` and later `result_of` attempt to call out to their function-type based interfaces. Since these are implementation details and only meant to be used immediately below their declarations, `_impl` would work just fine.